From a5363a8957efa2eb9f0feafe395033588b43e3c4 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Fri, 18 Dec 2015 18:46:32 +0800
Subject: [PATCH 0643/1110] drm/amd/dal: destroy mst connector after reset mode

MST semaphore lock/unlock done after actual dc_commit_target is
done. Need to make sure MST connectors alive till that time

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index e142508..33fe1a4 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -2244,8 +2244,6 @@ int amdgpu_dm_atomic_commit(
 		} /* switch() */
 	} /* for_each_crtc_in_state() */
 
-	update_connector_sem_state(state);
-
 	commit_targets_count = 0;
 
 	list_for_each_entry(crtc, &dev->mode_config.crtc_list, head) {
@@ -2261,6 +2259,8 @@ int amdgpu_dm_atomic_commit(
 	/* DC is optimized not to do anything if 'targets' didn't change. */
 	dc_commit_targets(dm->dc, commit_targets, commit_targets_count);
 
+	update_connector_sem_state(state);
+
 	/* update planes when needed */
 	for_each_plane_in_state(state, plane, old_plane_state, i) {
 		struct drm_plane_state *plane_state = plane->state;
-- 
2.7.4

